apps/src/applab/designElements/ImagePickerPropertyRow.jsx

```diff
@@ -1,6 +1,13 @@
 var showAssetManager = require('../assetManagement/show.js');
 var rowStyle = require('./rowStyle');
 
+// We'd prefer not to make GET requests every time someone types a character.
+// This is the amount of time that must pass between edits before we'll do a GET
+// I expect that the vast majority of time, people will be copy/pasting URLs
+// instead of typing them manually, which will result in an immediate GET,
+// unless they pasted within WAIT_TIME ms of editing the field manually
+const WAIT_TIME = 1500;
```

Comment:
user_input_delay, maybe? If the constant name is scoped to this file, I'm not too worried.